Libplanet/Blockchain/BlockChain.ProposeBlock.cs

```diff
@@ -145,11 +145,10 @@ public partial class BlockChain
             BlockHash prevHash = Store.IndexBlockHash(Id, index - 1)
                 ?? throw new NullReferenceException($"Chain {Id} is missing block #{index - 1}");
 
-            HashDigest<SHA256> stateRootHash =
-                _blocks[prevHash].ProtocolVersion <
-                BlockMetadata.SlothProtocolVersion
-                    ? _blocks[prevHash].StateRootHash
-                    : (HashDigest<SHA256>)GetNextStateRootHash(prevHash, TimeSpan.Zero);
+            HashDigest<SHA256> stateRootHash = GetNextStateRootHash(prevHash) ??
+                throw new InvalidOperationException(
+                    $"Cannot propose a block as the next state root hash " +
+                    $"for block {prevHash} is missing.");
```

Comment:
Since we've already waited for NSRH from setting validator set on consensus, we can assume this won't throw on the consensus, right?

Reply:
In essence, `ProposeBlock()` should only be called by consensus controller as a proposer. This should never be thrown if the PBFT algorithm is working as intended.